tools/dds/dds-server/lrs-device-controller.cpp

```diff
@@ -378,28 +449,58 @@ lrs_device_controller::lrs_device_controller( rs2::device dev, std::shared_ptr<
         throw std::runtime_error( "Empty dds_device_server" );
 
     _dds_device_server->on_open_streams( [&]( const json & msg ) { start_streaming( msg ); } );
-    _dds_device_server->on_close_streams( [&]( const json & msg ) { stop_streaming( msg ); } );
     _dds_device_server->on_set_option( [&]( const std::shared_ptr< realdds::dds_option > & option, float value ) {
         set_option( option, value );
     } );
     _dds_device_server->on_query_option( [&]( const std::shared_ptr< realdds::dds_option > & option ) -> float {
         return query_option( option );
     } );
 
-    //query_sensors returns a copy of the sensors, we keep it to use same copy throughout the run time.
-    //Otherwise problems could arise like opening streams and they would close at start_streaming scope end.
-    _sensors = _rs_dev.query_sensors();
-
     _device_sn = _rs_dev.get_info( RS2_CAMERA_INFO_SERIAL_NUMBER );
     LOG_DEBUG( "LRS device manager for device: " << _device_sn << " created" );
 
     // Create a supported streams list for initializing the relevant DDS topics
     auto supported_streams = get_supported_streams();
-
+    _bridge.on_start_sensor(
+        [this]( std::string const & sensor_name, dds_stream_profiles const & active_profiles )
+        {
+            auto & sensor = _rs_sensors[sensor_name];
```

Comment:
Might create sensor. Need to check if exist?

Reply:
I think we can pretty much count on the sensors not changing, as the streams do not change.
It'll complicate the syntax and readability, but I can add -- should I?

Reply:
No. I noticed now that sensor_name is assigned by the sensor bridge calling the callback function and it uses the already set sensor names.